app/screens/channel_files/channel_files.tsx

```diff
@@ -98,7 +100,18 @@ function ChannelFiles({
         const t = Date.now();
         lastSearchRequest.current = t;
         const searchParams = getSearchParams(channel.id, searchTerm, ftr);
-        const {files} = await searchFiles(serverUrl, channel.teamId, searchParams);
+        let files: FileInfo[] | undefined = [];
+        if (channel.teamId) {
+            const searchedFiles = await searchFiles(serverUrl, channel.teamId, searchParams);
+            files = searchedFiles.files;
+        } else {
+            // teamId can be empty for GMs and DMs, as do not store teamId for GMs and DMs in the database
```

Comment:
Nitpicking, but the team ID should always be empty for DMs and GMs. If it's ever not, then we have a problem. Or likely multiple problems 😅